backend/routes/users.js

```diff
@@ -0,0 +1,58 @@
+import express from "express";
+import { User } from "../models/user.js";
+import { Op } from "sequelize";
+import bcrypt from "bcrypt";
+
+const router = express.Router();
+
+// Creating a new user in database
+router.post("/users/signup", async (req, res) => {
+    const { username, password, email } = req.body;
+
+    try {
+        const userAlreadyExists = await User.findOne({
+            where: { [Op.or]: [{ username }, { email }] }
+        });
+
+        if (userAlreadyExists) {
+            return res.status(400).json({ error: "Username or email already exists." })
+        }
+
+        const hashedPassword = await bcrypt.hash(password, 10);
+
+        const newUser = await User.create({ username, password: hashedPassword, email });
+
+        req.session.user = newUser;
+
+        res.status(200).json({ user: newUser })
+    } catch (error) {
+        res.status(500).json({ error: "Server error." });
+    }
+});
+
+router.post("/users/login", async (req, res) => {
+    const { username, password } = req.body;
+
+    const user = await User.findOne({ where: { username } });
```

Comment:
Nit
This is all fine. I think there is an alternative with less code, you can query for something 
WHERE
username = username
password = hashedPassword

In this case you would be hashing the password in the params before trying to find the user.